### ISSUE TITLE
setting up OSC ports

### ISSUE THREAD

**Issue by anjtb** [CINPLA/tracking-plugin#16]
I don't seem to be able to edit the OSC ports. How do I get to the TransportConfiguration Collection Editor window? 

The Wiki suggests there is a script called osc.bonsai, but I can't locate it.

Sorry if this is an obvious question!

**Comment by alejoe91** [comment]
Hi,

The `osc.bonsai` is in the Bonsai folder (https://github.com/CINPLA/tracking-plugin/tree/master/Bonsai).

Cheers
Alessio

**Comment by anjtb** [comment]
Ah, Ok. It has a .config extension - maybe that's why I was confused. Do I use this .config file in the same way as a .bonsai file?

**Comment by alejoe91** [comment]
It should be automatically detected by the bonsai script.

**Comment by anjtb** [comment]
Ah, OK. 

I still can't get to the TransportConfiguration Collection Editor window shown on the Wiki. 

I also couldn't find the Osc Design Library package- I was wondering if this was the problem?

**Comment by alejoe91** [comment]
Yes I think it could be the problem. Also we haven't really tested with Bonsai>=2.3, so there might have been some design changes. Since this is Bonsai-related, maybe a good place to ask would be the Bonsai forum or gitter channel!

**Comment by hwpdalgleish** [comment]
Hi both, I found this thread while trying to solve a similar issue (using Bonsai 2.6.2) and have subsequently solved it so thought I'd post if useful. @alejoe91 is right - Bonsai has changed the way you set up OSC channels so you no longer need to use the config script. Instead you need to:

- set up two udp clients at the top of the workflow (search udp in the "search modules" bar in the Toolbox window on the left of the Bonsai window and create two of them). One is going to be the red port, the other the green port. Click on each and input the relevant details in the Properties window on the right (Name, Port, RemoteHostName and RemotePort - these are the same values as depicted in the screenshots of the original Bonsai workflow here https://github.com/CINPLA/tracking-plugin/wiki). See below my example of the right-hand side of the Bonsai workflow demonstrating this for the RedPort (note it is highlighted white):
![OSC_setup](https://user-images.githubusercontent.com/15194455/112965464-30a5b500-9141-11eb-820a-751f8a0cd03f.PNG)

- connect each SendMessage module to its corresponding udp client by clicking on each and entering the corresponding Address and Connection in the Properties window on the right. See below for the SendMessage module connected to the RedPort (note it is highlighted white):
![OSC_output](https://user-images.githubusercontent.com/15194455/112965931-a3169500-9141-11eb-925c-d3bca483a09f.PNG)

Now you should be able to add the Tracking Port and Tracking Visual and input the corresponding Port and Address for each SendMessage+udp in the Tracking Port to pull in the data from Bonsai.

Hope this helps!

**Comment by alejoe91** [comment]
Hi @hwpdalgleish 

Thank you very much for this!!! I'll update the wiki accordingly :)

Happy Easter!
Alessio

**Comment by pplenck** [comment]
Hello, 
now in the 2026 version of bonsai, I cannot get the send message to link with the redport udp client. 
I get a runtime error: 'the specified connection name has no matching configuration'.
Can we get more info on how these two work . Something is wrong between the ports qnd the send messqge modules. or maybe i's the data format that is not arranged properly?

<img width="685" height="455" alt="Image" src="https://github.com/user-attachments/assets/0593911b-ba26-4545-a37f-4500a75b1041" />
<img width="773" height="455" alt="Image" src="https://github.com/user-attachments/assets/9cd7dc12-140c-429c-a1ab-c4b213066828" />

**Comment by alejoe91** [comment]
Hi @pplenck 

Not sure how to update this. Could you open an issue on the bonsai/open ephys page? They have taken over the maintenance :)
